pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/blocks/TransferableBlockUtils.java

```diff
@@ -84,7 +83,8 @@ public static Object[] getRow(TransferableBlock transferableBlock, int rowId) {
     if (transferableBlock.isContainerBlock() && transferableBlock.getType() == BaseDataBlock.Type.ROW) {
       return transferableBlock.getContainer().get(rowId);
     } else {
-      return SelectionOperatorUtils.extractRowFromDataTable(transferableBlock.getDataBlock(), rowId);
+      // TODO: optimize the roaring bitmap decoding to only be done once and reused across all row retrieval.
+      return DataBlockUtils.extractRow(transferableBlock.getDataBlock(), rowId);
```

Comment:
found a bug from #9618 this should fix it but not as efficient. need to think how to reuse the roaring null bitmap without duplicate extracting it.

Reply:
This bug is addressed and the efficiency issue fixed as well